### PR TITLE
#855: Ban users without requiring them to be in the server

### PR DESCRIPTION
**Pull request by butteredwaffles**
Based on some testing I did with a few friends, this should still purge the messages of banned users who left before they were banned.

### CODE REVIEW

**Comment by butteredwaffles** [comment]
yeah i'm not really sure what to do about the type errors here

**Review by Jay-Madden — DISMISSED** [review]
This is badly needed, some db work might be needed here tho

File: ClemBot.Bot/bot/services/moderation_service.py [hunk 1]
```diff
@@ -36,20 +36,21 @@ async def on_bot_ban(
...
-            raise_on_error=True,
-        )
+        if isinstance(subject, discord.Member):
+            await self.bot.moderation_route.insert_ban(
+                guild_id=guild.id,
+                author_id=author.id,
+                subject_id=subject.id,
+                reason=reason,
+                raise_on_error=True,
+            )
```
[the remark this answers is not in the document]
Comment (butteredwaffles) [reply ClemBot.Bot/bot/services/moderation_service.py:53]: Ah - I'm not as familiar with navigating large C# projects like the API. Where's the SQL script located that needs to be updated / would I have to add a migration script somewhere to drop the constraint?

also - would the fixing problem with the /infractions list command just doing the same thing I did here with converting ints to users for viewing?

**Review by jkriste — APPROVED** [review]
LGTM! 🎉

**Pull request merged**